VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/SharedComponents/AppointmentCancelReschedule.tsx

```diff
@@ -140,6 +154,218 @@ const cancelButton = (
   )
 }
 
+const canceledPending = (location: AppointmentLocation | undefined, theme: VATheme, t: TFunction) => {
+  return (
+    <Box>
+      {spacer(theme)}
+      <TextView variant="MobileBodyBold" accessibilityRole="header" mt={theme.dimensions.standardMarginBetween}>
+        {t('appointments.reschedule.pending.title')}
+      </TextView>
+      <TextView
+        variant="MobileBody"
+        mb={theme.dimensions.condensedMarginBetween}
+        accessibilityLabel={a11yLabelVA(t('appointments.reschedule.pending.body'))}>
+        {t('appointments.reschedule.pending.body')}
+      </TextView>
+      {location?.phone && location.phone.areaCode && location.phone.number ? (
+        <ClickToCallPhoneNumber phone={location.phone} />
+      ) : undefined}
+      <LinkWithAnalytics
+        type="url"
+        url={LINK_URL_VA_SCHEDULING}
+        text={t('appointments.vaSchedule')}
+        a11yLabel={a11yLabelVA(t('appointments.vaSchedule'))}
+      />
+    </Box>
+  )
+}
+
+const inPersonPhoneVideoVACanceled = (location: AppointmentLocation | undefined, theme: VATheme, t: TFunction) => {
+  return (
+    <Box>
+      {spacer(theme)}
```

Comment:
Looks like almost all of this is repeated 6 times. Wondering if it can be refactored into a function with some arguments, or split up into a few smaller reusable functions

Reply:
refactored